codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketClientHandshaker.java

```diff
@@ -431,7 +464,50 @@ public ChannelFuture close(Channel channel, CloseWebSocketFrame frame, ChannelPr
         if (channel == null) {
             throw new NullPointerException("channel");
         }
-        return channel.writeAndFlush(frame, promise);
+        channel.writeAndFlush(frame, promise);
+        applyForceCloseTimeout(channel, promise);
+        return promise;
+    }
+
+    private void applyForceCloseTimeout(final Channel channel, ChannelFuture flushFuture) {
+        final long forceCloseTimeoutMillis = this.forceCloseTimeoutMillis;
+        if (forceCloseTimeoutMillis <= 0 || !channel.isActive()) {
```

Comment:
nit: you may also want to check `!forceCloseInit.get()`

Reply:
nice!